### PR TITLE
EW-12480 Remove double quotes to string values in EdgeKV CLI 

### PR DESCRIPTION
**Pull request by kyleli-akamai**
### Major Change
- Fix issues that double quotes will be added to the string values in backend
- Add unit tests for ekv-services
- Bump up the CLI version to `1.5.2`

### Test Results (See `testing writeItems` case):
```
jest ./tests/edgekv                                                                        
 PASS  tests/edgekv/ekv-services.test.ts
  ekv-services tests
    testing getNameSpaceList
      ✓ URL path should not include query parameter when details parameter is false (1 ms)
      ✓ URL path should contian query parameter when details parameter is true (1 ms)
      ✓ function should handle errors properly
    testing getGroupsList
      ✓ response should return groups list
      ✓ function should handle errors properly
    testing createNameSpace
      ✓ response should return created namespace (1 ms)
      ✓ function should handle errors properly
    testing getNameSpace
      ✓ response should return namespace info (1 ms)
      ✓ function should handle errors properly
    testing updateNameSpace
      ✓ response should return changed namespace info (1 ms)
      ✓ function should handle errors properly
    testing initializeEdgeKV
      ✓ response should return initialization success
      ✓ function should handle errors properly (1 ms)
    testing getInitializeEdgeKV
      ✓ response should return initialization success (2 ms)
      ✓ function should handle errors properly
    testing wrtieItems
      ✓ URL path should not contain sandboxId if the query parameter is not provided (1 ms)
      ✓ URL path should contain sandboxId if the query parameter is provided
      ✓ response should return writeItems success when body is a JSON object string
      ✓ function should handle errors properly
    testing wrtieItemsFromFile
      ✓ URL path should not contain sandboxId if the query parameter is not provided (1 ms)
      ✓ URL path should contain sandboxId if the query parameter is provided (1 ms)
      ✓ function should handle errors properly
    testing readItem
      ✓ URL path should not contain sandboxId if the query parameter is not provided (1 ms)
      ✓ URL path should contain sandboxId if the query parameter is provided
      ✓ function should handle errors properly
    testing deleteItem
      ✓ URL path should not contain sandboxId if the query parameter is not provided (1 ms)
      ✓ URL path should contain sandboxId if the query parameter is provided
      ✓ function should handle errors properly
    testing getItemsFromGroup
      ✓ URL path should not contain maxItems if the query parameter is not provided
      ✓ URL path should contain maxItems if the query parameter is provided
      ✓ URL path should not contain sandboxId if the query parameter is not provided
      ✓ URL path should contain sandboxId if the query parameter is provided
      ✓ URL path should contain maxItems and sandboxId if both query parameters are provided
      ✓ function should handle errors properly
    testing createEdgeKVToken
      ✓ response should return creation success
      ✓ function should handle errors properly
    testing getSingleToken
      ✓ response should return single token success
      ✓ function should handle errors properly
    testing getTokenList
      ✓ URL path should not include includeExpired query parameter if passing false
      ✓ URL path should include includeExpired query parameter if passing true (1 ms)
      ✓ function should handle errors properly (1 ms)
    testing revokeToken
      ✓ response should revoke token success
      ✓ function should handle errors properly
    testing modifyAuthGroupPermission
      ✓ response should modify permissions successfully (1 ms)
      ✓ function should handle errors properly
    testing listAuthGroups
      ✓ URL path should not include groupId when passing 0
      ✓ URL path should include include groupId when passing non-zero value
      ✓ function should handle errors properly (1 ms)

Test Suites: 1 passed, 1 total
Tests:       48 passed, 48 total
Snapshots:   0 total
Time:        1.088 s
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "akamai-edgeworkers-cli",
-  "version": "1.5.1",
+  "version": "1.5.2",
```
Comment (bmatthew) [inline package.json:3]: I appreciate you updating the version number here. I had marked the tickets for this release as 1.6.0 but we can go with 1.5.2 as it's a bugfix update. I'll update the tickets in JIRA.

File: src/edgekv/ekv-service.ts [hunk 2]
```diff
@@ -10,121 +10,336 @@ const INIT_EKV_TIMEOUT = 120000;
...
   let body = itemList;
   let writeItemPath = `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`;
-  if(sandboxid){
-    writeItemPath += `?sandboxId=${sandboxid}`
+  if (sandboxid) {
+    writeItemPath += `?sandboxId=${sandboxid}`;
+  }
+
+  let request: Promise<any>;
+  if (typeof body === 'string') {
```
Comment (bmatthew) [inline src/edgekv/ekv-service.ts:133]: Is this the section that fixes the bug in EW-12480?

It's good to see some clean up done as well but I just want to confirm.

File: tests/edgekv/ekv-services.test.ts [hunk 3]
```diff
@@ -0,0 +1,1421 @@
+import * as httpEdge from '../../src/cli-httpRequest';
+import * as ekvService from '../../src/edgekv/ekv-service';
+import { ekvMetrics } from '../../src/edgekv/ekv-metricFactory';
+
+describe('ekv-services tests', () => {
```
Comment (bmatthew) [inline tests/edgekv/ekv-services.test.ts:5]: This is a very thorough test!

**Review by bmatthew — APPROVED** [review]

File: src/edgekv/ekv-service.ts [hunk 4]
```diff
@@ -10,121 +10,336 @@ const INIT_EKV_TIMEOUT = 120000;
...
+      body,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.modifyAuthGroup
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
+// TODO: Should `groupId` parameter be optional?
 export function listAuthGroups(groupId: number) {
```
Comment (kyleli-akamai) [inline src/edgekv/ekv-service.ts:332]: Hi @bmatthew, I wonder if this `groupId` parameter should be optional.

Comment (bmatthew) [reply src/edgekv/ekv-service.ts:332]: Yeah it should be. The call into here in listAuthGroups in ekv-handler (line 349) seems like it works around the optionality missing by calling with a variable assigned to null so if you change this you could modify that as well to just call with no variable

File: src/edgekv/ekv-service.ts (hunk 2 above)
Comment (kyleli-akamai) [reply src/edgekv/ekv-service.ts:133]: Yes, this is the fix.

File: src/edgekv/ekv-service.ts (hunk 4 above)
Comment (kyleli-akamai) [reply src/edgekv/ekv-service.ts:332]: Do we want to handle some invalid cases like passing non-positive numbers?

Comment (bmatthew) [reply src/edgekv/ekv-service.ts:332]: Invalid options should already be handled by the OPEN API called here so it shouldn't be needed.

**Pull request merged**